engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugOverlay.java

```diff
@@ -111,10 +122,25 @@ public String get() {
                     Vector3i chunkPos = ChunkMath.calcChunkPos((int) pos.x, (int) pos.y, (int) pos.z);
                     Vector3f rotation = localPlayer.getViewDirection();
                     Vector3f cameraPos = localPlayer.getViewPosition();
-                    return String.format(Locale.US, "Position: (%.2f, %.2f, %.2f), Chunk (%d, %d, %d), Eye (%.2f, %.2f, %.2f), Rot (%.2f, %.2f, %.2f)", pos.x, pos.y, pos.z,
+
+                    //compass direction debugging
+                    String compassDir = "N";
+                    double highEq=0.4;
+                    compassDir = rotation.x > 0.0-highEq && rotation.x < 0.0+highEq && rotation.z<0 && rotation.z>-1 ? "N" : compassDir;
+                    compassDir = rotation.x > 0.6-highEq && rotation.x < 0.6+highEq && rotation.z<0 && rotation.z>-1 ? "NE" : compassDir;
+                    compassDir = rotation.z > 0.0-highEq && rotation.z < 0.0+highEq && rotation.x<1 && rotation.x>0 ? "E" : compassDir;
+                    compassDir = rotation.z > 0.6-highEq && rotation.z < 0.6+highEq && rotation.x<1 && rotation.x>0 ? "SE" : compassDir;
+                    compassDir = rotation.x > 0.0-highEq && rotation.x < 0.0+highEq && rotation.z<1 && rotation.z>0 ? "S" : compassDir;
+                    compassDir = rotation.x > -0.6-highEq && rotation.x < -0.6+highEq && rotation.z<1 && rotation.z>0 ? "SW" : compassDir;
+                    compassDir = rotation.z > 0.0-highEq && rotation.z < 0.0+highEq && rotation.x<0 && rotation.x>-1 ? "W" : compassDir;
+                    compassDir = rotation.x > -0.6-highEq && rotation.x < -0.6+highEq && rotation.z<0 && rotation.z>-1 ? "NW" : compassDir;
```

Comment:
Is it ever possible that x or z have a magnitude larger than 1? It might also be easier to read if you assign them to shorter variables `rotX` and `rotZ`.